### PR TITLE
Mutation test

### PR DESCRIPTION
**Pull request by vakepati**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-283

### Change description ###

Added Mutation Testing coverage

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -121,11 +122,45 @@ jacocoTestReport {
...
+    'VOID_METHOD_CALLS',
+    'EMPTY_RETURNS',
+    'FALSE_RETURNS',
+    'NULL_RETURNS',
+    'PRIMITIVE_RETURNS',
+    'TRUE_RETURNS',
+    'REMOVE_CONDITIONALS'
+  ])
+}
+project.tasks['pitest'].group = "Verification"
```
Comment (doncem) [inline build.gradle:155]: i guess `group` can go straight into task definition itself

File: build.gradle [hunk 2]
```diff
@@ -159,6 +194,7 @@ def versions = [
   junit: '5.6.2',
   junitPlatform: '1.6.2',
   mockitoJupiter: '3.3.3',
+  springBoot: springBoot.class.package.implementationVersion,
```
Comment (doncem) [inline build.gradle:197]: no need for that

File: build.gradle [hunk 3]
```diff
@@ -5,10 +5,11 @@ plugins {
   id 'jacoco'
   id 'io.spring.dependency-management' version '1.0.9.RELEASE'
   id 'org.flywaydb.flyway' version '6.4.3'
-  id 'org.springframework.boot' version '2.3.0.RELEASE'
+  id 'org.springframework.boot' version '2.2.7.RELEASE'
```
Comment (doncem) [inline build.gradle:8]: wait a sec? why the downgrade?

File: build.gradle [hunk 4]
```diff
@@ -174,6 +210,19 @@ dependencyManagement {
...
+      entry 'log4j-api'
+    }
+    //dom4j-2.1.1.jar cause CVE-2020-10683. remove when spring-boot-starter-data-jpa uses dom 2.1.3
+    dependency group: 'org.dom4j', name: 'dom4j', version: '2.1.3'
+
+    // CVE-2020-9484 present in 9.0.34
+    dependencySet(group: 'org.apache.tomcat.embed', version: '9.0.36') {
+      entry 'tomcat-embed-core'
+      entry 'tomcat-embed-websocket'
+    }
```
Comment (doncem) [inline build.gradle:225]: with latest spring boot all this should go away

File: build.gradle (hunk 3 above)
Comment (vakepati) [reply build.gradle]: upgrade done

File: gradle/wrapper/gradle-wrapper.properties [hunk 5]
```diff
@@ -1,5 +1,6 @@
+#Mon Jun 08 23:41:34 BST 2020
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.4.1-all.zip
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.4.1-bin.zip
-zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
```
Comment (rkondratowicz) [inline gradle/wrapper/gradle-wrapper.properties:6]: can be reverted

File: build.gradle [hunk 6]
```diff
@@ -249,6 +284,10 @@ dependencies {
   testCompile group: 'com.icegreen', name: 'greenmail', version: '1.5.13'
   testCompile group: 'org.apache.commons', name: 'commons-email', version: '1.5'
 
+  testCompile group: 'org.pitest', name: 'pitest', version: '1.5.1'
+  testCompile 'info.solidsoft.gradle.pitest:gradle-pitest-plugin:1.5.1'
+  testCompile 'org.codehaus.sonar-plugins:sonar-pitest-plugin:0.5'
```
Comment (rkondratowicz) [inline build.gradle:289]: let's be consistent and use the map notation here as well

**Comment by doncem** [comment]
There's a vulnerability reported against newly added libraries:

```
> Task :dependencyCheckAggregate
Generating report for project bulk-scan-processor
Found 3 vulnerabilities in project bulk-scan-processor

One or more dependencies were identified with known vulnerabilities in bulk-scan-processor:

gradle-pitest-plugin-1.5.1.jar (cpe:2.3:a:gradle:gradle:1.5.1:*:*:*:*:*:*:*) : CVE-2019-11065, CVE-2019-15052, CVE-2019-16370
```

File: build.gradle [hunk 7]
```diff
@@ -249,6 +284,10 @@ dependencies {
   testCompile group: 'com.icegreen', name: 'greenmail', version: '1.5.13'
   testCompile group: 'org.apache.commons', name: 'commons-email', version: '1.5'
 
+  testCompile group: 'org.pitest', name: 'pitest', version: '1.5.1'
+  testCompile group: 'info.solidsoft.gradle.pitest', name: 'gradle-pitest-plugin', version: '1.5.1'
```
Comment (rkondratowicz) [inline build.gradle:288]: is this needed? the plugin is declared in the 'plugins' section

File: gradle/wrapper/gradle-wrapper.properties [hunk 8]
```diff
@@ -1,3 +1,4 @@
+#Mon Jun 08 23:41:34 BST 2020
```
Comment (rkondratowicz) [inline gradle/wrapper/gradle-wrapper.properties:1]: can be reverted

**Pull request closed without merging**